src/runtime/database.cpp

```diff
@@ -883,11 +896,41 @@ void Database::insert_job(const std::string &directory, const std::string &comma
   end_txn();
 }
 
+template <class F>
+static void scan_until_sep(char sep, const std::string &to_scan, F f) {
+  auto begin = to_scan.begin();
+  auto cur_start = begin;
+  auto end = to_scan.end();
+  while (begin < end) {
+    if (*begin == sep) {
+      // TODO: This would be better as a string_view. Since the allocation
+      //       would not be forced. To compremise we can use an r-value string
```

Comment:
spelling; compremise

Reply:
This handles the empty string differently. I'd check that this does not matter.